flake.nix

```diff
@@ -38,447 +34,456 @@
-            --execution-jwt ${options.lighthouse.datadir}/jwt.hex \
-            --builder ${options.lighthouse.mev-boost.endpoint} \
-            --prune-payloads false \
-            --metrics \
-            ${if options.lighthouse.slasher.enable then
-              "--slasher "
-              + " --slasher-history-length " + (toString options.lighthouse.slasher.history-length)
-              + " --slasher-max-db-size " + (toString options.lighthouse.slasher.max-db-size)
-            else "" }
-          '';
-              wantedBy = [ "multi-user.target" ];
             };
+          };
 
-            # firewall
-            networking.firewall = {
-              allowedTCPPorts = [ 9000 ];
-              allowedUDPPorts = [ 9000 ];
-              interfaces."wg0".allowedTCPPorts = [
-                5052 # lighthouse
+          homestakeros = {
+            system = "x86_64-linux";
+            specialArgs = { inherit inputs outputs; };
+            modules = [
+              ./system
+              ./system/ramdisk.nix
+              ./system/formats/netboot-kexec.nix
+              self.nixosModules.homestakeros
+              home-manager.nixosModules.home-manager
+              disko.nixosModules.disko
+              {
+                nixpkgs.overlays = [
+                  ethereum-nix.overlays.default
+                  outputs.overlays.additions
+                  outputs.overlays.modifications
+                ];
+                system.stateVersion = "23.05";
+              }
+            ];
+          };
+        in
+        rec {
+          # filters options recursively
+          # option exports -- accessible through 'nix eval --json .#exports'
+          exports = nixpkgs.lib.attrsets.mapAttrsRecursiveCond
+            (v: ! nixpkgs.lib.options.isOption v)
+            (k: v: v.type.name)
+            homestakeros_options;
+
+          overlays = import ./overlays { inherit inputs; };
+
+          nixosConfigurations = with nixpkgs.lib; {
+            "homestakeros" = nixosSystem (getAttrs [ "system" "specialArgs" "modules" ] homestakeros);
+          } // (with nixpkgs-stable.lib; { });
+
+          # usage: https://github.com/ponkila/homestaking-infra/commit/574382212cf817dbb75657e9fef9cdb223e9823b
+          nixosModules.homestakeros = { config, lib, pkgs, ... }: with nixpkgs.lib;
+            let
+              cfg = config.homestakeros;
+            in
+            {
+              options.homestakeros = homestakeros_options;
+
+              config = mkMerge [
+                ################################################################### LOCALIZATION
+                (mkIf true {
+                  networking.hostName = cfg.localization.hostname;
+                  time.timeZone = cfg.localization.timezone;
+                })
+
+                #################################################################### MOUNTS
+                (mkIf true {
```

Comment:
Also, I think that the module option mounts are still broken. 😮‍💨 

Nix repl (main):
```
[I] kari@torque ~/D/t/homestaking-infra (main) $ nix repl
Welcome to Nix 2.15.0. Type :? for help.

nix-repl> :lf .#
Added 22 variables.

nix-repl> outputs.nixosConfigurations.ponkila-ephemeral-beta.config.systemd.mounts
[ { ... } { ... } { ... } { ... } ]
```


Nix repl (this PR):
```
[I] kari@hetzner-ephemeral-alpha ~/homestaking-infra (juuso/config-extraction)> nix repl
Welcome to Nix 2.13.3. Type :? for help.

nix-repl> :lf .#
Added 22 variables.

nix-repl> outputs.nixosConfigurations.ponkila-ephemeral-beta.config.systemd.mounts
[ { ... } ]
```